src/libraries/System.IO.Compression/src/System/IO/Compression/Tar/TarHeader.cs

```
@@ -0,0 +1,667 @@
+        // Throws if end of stream is reached or if a conversion to an expected data type fails.
+        private void ReadUstarAttributes(Stream archiveStream)
+        {
+            _blocks.ReadPosixPrefixAttributeBytes(archiveStream);
+
+            Prefix = TarHelpers.GetTrimmedUtf8String(_blocks._prefixBytes);
+
+            // In ustar, Prefix is used to store the *leading* path segments of
+            // Name, if the full path did not fit in the Name byte array.
+            if (!string.IsNullOrEmpty(Prefix))
+            {
+                Name = Path.Join(Prefix, Name);
+            }
+
+            _blocks.ReadPosixPaddingBytes(archiveStream);
+        }
+
+        // Collects the extended attributes found in the data section of a PAX entry of type 'x' or 'g'.
+        // Throws if end of stream is reached or if an attribute is malformed.
+        private void ReadPaxExtendedAttributes(Stream archiveStream)
+        {
+            Debug.Assert(TypeFlag is TarEntryTypeFlag.ExtendedAttributes or TarEntryTypeFlag.GlobalExtendedAttributes);
+
+            if (Size > 0)
+            {
+                // Highly doubtful that a long path will be longer than int.MaxValue,
+                // considering 4096 is a common max path length.
+                Debug.Assert(Size <= int.MaxValue);
+
+                _extendedAttributes ??= new();
+
+                byte[] buffer = new byte[(int)Size];
+                if (archiveStream.Read(buffer.AsSpan()) != Size)
+                {
+                    throw new EndOfStreamException();
+                }
+
+                string longPath = TarHelpers.GetTrimmedUtf8String(buffer);
+
+                using StringReader reader = new(longPath);
+
+                while (TryGetNextExtendedAttribute(reader, out string? key, out string? value))
+                {
+                    if (!string.IsNullOrEmpty(key) && !string.IsNullOrEmpty(value) && !_extendedAttributes.ContainsKey(key))
```

This silently skips duplicate attributes. Is this the intended behavior?

---

I don't think the spec mentions anything about duplicate attributes. But I'm pretty sure that's unexpected behavior. Good catch, I think I'll throw instead.